astropy/convolution/setup_package.py

```diff
@@ -17,8 +17,8 @@ def get_extensions():
     # Add '-Rpass-missed=.*' to ``extra_compile_args`` when compiling with clang
     # to report missed optimizations
     sources = [
-        os.path.join(C_CONVOLVE_PKGDIR, "_convolve.pyx"),
-        os.path.join(C_CONVOLVE_PKGDIR, "src", "convolve.c"),
+        str(C_CONVOLVE_PKGDIR / "_convolve.pyx"),
+        str(C_CONVOLVE_PKGDIR / "src" / "convolve.c"),
```

Comment:
It would be nice if setuptools supported Paths for path-like things. Lot of conversations in this PR, did you submit a PR to setuptools? Either way, I'm fine to with this.

Reply:
Not yet! I will send one. I did create one for astropy's extension-helpers https://github.com/astropy/extension-helpers/pull/83

Reply:
https://github.com/pypa/distutils/pull/230 is up too :)